url27db8982cd_leohinterlang_argFace_tgz-pJ8-com_fidelis_argface_ArgSampleJ8/src/com/fidelis/argface/ArgParseUsage.java

```diff
@@ -0,0 +1,754 @@
+    private List<ArgOperand>     varList    = new ArrayList<ArgOperand>();
+    private List<ArgOperand>     litList    = new ArrayList<ArgOperand>();
+    private List<ArgOption>      optionList = new ArrayList<ArgOption>();
+    
+    /**
+     * Parses the specified usage text.
+     * 
+     * @param usage the usage text
+     * @return {@code true} if successful
+     */
+    public boolean parse (String usage) {
+        Debug.trace(usage);
+        boolean status = true;
+        source = new TokenString(usage, delimiters);
+        argList = new ArgList();
+        
+        // First token. Usage is optional.
+        String token = source.next();
+        if (token.equalsIgnoreCase("usage")) {
+            token = source.next();
+        }
+        if (token.equals(":")) {
+            token = source.next();
+        }
+        programName = token;
+        
+        // Each time program name occurs is another use spec.
+        while (token.equals(programName)) {
+            operand = new ArgOperand().literal();
+            operand.setName(programName);
+            argList.goHome();
+            argList.endSouth();
+            argList.addSouth(operand);
+            token = parseUseSpec();
+        }
+        if ("options".equalsIgnoreCase(token)) {
+            if (! parseOptionsSection()) {
+                return false;
+            }
+        }
+        if ("error".equals(token)) {
+            return false;
+        }
+        for (ArgOperand operand : varList) {
+            Debug.trace("np var operand: " + operand);
+        }
+        for (ArgOperand operand : litList) {
+            Debug.trace("np lit operand: " + operand);
+        }
+        for (ArgOption option : optionList) {
+            Debug.trace("np     option:  " + option);
+        }
+        Debug.trace(argList.toString());
+        String u = argList.buildUsage();
+        Debug.trace(u);
+        return status;
+    }
+    
+    /**
+     * Returns the program name as specified in the usage text.
+     * 
+     * @return the program name from the usage text
+     */
+    public String getProgramName () {
+        return programName;
+    }
+    
+    /**
+     * Returns the argument list.
+     * 
+     * @return the argument list
+     */
+    public ArgList getArgList () {
+        return argList;
+    }
+    
+    /**
+     * Returns the variable operands list.
+     * 
+     * @return the variable operands list
+     */
+    public List<ArgOperand> getVarList () {
+        return varList;
+    }
+    
+    /**
+     * Returns the literal operands list.
+     * 
+     * @return the literal operands list
+     */
+    public List<ArgOperand> getLitList () {
+        return litList;
+    }
+    
+    /**
+     * Returns the option list.
+     * 
+     * @return the option list
+     */
+    public List<ArgOption> getOptionList () {
+        return optionList;
+    }
+    
+    private String parseUseSpec () {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 22. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.